complete.sh

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env bash
+# Source this file and then...
+
+# TODO: use after running the function
+compopt () {
```

Comment:
I *think* what you can do is have a completion be something like `complete -o dirs -F _my_fancy_func`. Not sure if it merges the result or uses one as the filter... Anyway, inside of `_my_fancy_func` you can call e.g. `compopt -o noquote` to "update" the `complete` command. If no completion with `compgen` or similar is going on, the builtin `compopt` will hard fail, that's why I made it a noop.

TBH I'm quite puzzled as to why the `compgen` command warns that `-F` and `-C` completions may not give back expected results instead of just assuming the cursor to be at the end of the line. It seems to do virtually no bootstrapping as done during an actual autocompletion... I really had to stop myself from rewriting all of `compgen` in pure bash 😄